monai/inferers/inferer.py

```diff
@@ -64,6 +67,190 @@ def __call__(self, inputs: torch.Tensor, network: Callable, *args: Any, **kwargs
+        batch_size: batch size for patches. If the input tensor is already batched [BxCxWxH],
+            this adds additional batching [(Bp*B)xCxWpxHp] for inference on patches. Defaults to 1.
+        pre_processor: a callable that process patches before the being fed to the network. Defaults to None.
+        post_processor: a callable that process the output of the network. Defaults to None.
+        output_keys: if the network output is a dictionary, this defines the keys of the output dictionary to use.
+            Defaults to None, where all the keys are taken if output is a dictionary.
+    """
+
+    def __init__(
+        self,
+        splitter: Splitter | None = None,
+        merger: Merger | Sequence[Merger] | None = None,
+        batch_size: int = 1,
+        pre_processor: Callable | None = None,
+        post_processor: Callable | None = None,
+        output_keys: Sequence | None = None,
+    ) -> None:
+        Inferer.__init__(self)
+
+        # splitter
+        if splitter is not None and not callable(splitter):
+            raise TypeError(f"'splitter' should be a callable object, {type(splitter)} is given.")
+        self.splitter = splitter
+
+        # mergers
+        self.mergers: Sequence[Merger] = (AvgMerger(),) if merger is None else ensure_tuple(merger)
+        for m in self.mergers:
+            if not isinstance(m, Merger):
+                raise TypeError(f"'merger' should be a `Merger` object, {type(m)} is given.")
+
+        # pre-processor (process patch before the network)
+        if pre_processor is not None and not callable(pre_processor):
+            raise TypeError(f"'pre_processor' should be a callable object, {type(pre_processor)} is given.")
+        self.pre_processor = pre_processor
+
+        # post-processor (process the output of the network)
+        if post_processor is not None and not callable(post_processor):
+            raise TypeError(f"'post_processor' should be a callable object, {type(post_processor)} is given.")
+        self.post_processor = post_processor
+
+        # batch size for patches
+        self.batch_size = batch_size
+
+        # model output keys
+        self.output_keys = output_keys
+
+    def _batch_sampler(
+        self, patches: Iterable[tuple[torch.Tensor, Sequence[int]]] | MetaTensor
+    ) -> Iterator[tuple[torch.Tensor, Sequence, int]]:
+        """Generate batch of patches and locations
+
+        Args:
+            patches: a tensor or list of tensors
+
+        Yields:
+            A batch of patches (torch.Tensor or MetaTensor), a sequence of location tuples, and the batch size
+        """
+        if isinstance(patches, MetaTensor):
+            total_size = len(patches)
+            for i in range(0, total_size, self.batch_size):
+                batch_size = min(self.batch_size, total_size - i)
+                yield patches[i : i + batch_size], patches[i : i + batch_size].meta[PatchKeys.LOCATION], batch_size  # type: ignore
```

Comment:
Do we expect every general patch MetaTensor to have the `LOCATION` field in the `meta`?
CC @wyli 

Thanks.

Reply:
It is not a requirement for using `PatchInferer` (for almost all use cases). This is only required if all the following are satisfied:
1) the data is already patched (for instance if the data is stored in patches on the disk)
2) so the user don't want to use any `Spliter` in `PatchInferer`
3) and the user wants to load all the patches into a `MetaTensor`.

In this case, there should be a standard way to have the MetaTensor of patches being represented, so the user should use `PatchKeys.LOCATION`.
Otherwise, the user still can define the input as a list of (MetaTensor, location), which is generic and does not need `PatchKeys.LOCATION`.

Reply:
OK, now I understand the logic.
Could you please help add some check here for `meta[PatchKeys.LOCATION]`? If users use `MetaTensor` to represent the data by mistake, better to give clear tips.

Thanks.

Reply:
I added the check and also unittest. Thanks